File: ios/StockPro/Data/DTOs/CustomerDTOs.swift
import Foundation

struct CustomerDTO: Decodable, Identifiable {
    let id: Int
    let name: String
    let type: String?
    let email: String?
    let phone: String?
    let address: String?
    let client_code: String?
    let discount_rate: Double?
    let is_loyal: Bool?
    let is_active: Bool?
    let ice: String?
    let notes: String?

    func toItem() -> CustomerListItem {
        CustomerListItem(
            id: id,
            name: name,
            clientCode: client_code ?? "",
            phone: phone ?? "",
            email: email ?? "",
            type: type ?? "particulier",
            isLoyal: is_loyal ?? false
        )
    }
}

struct CustomerCreateRequest: Encodable {
    let name: String
    let type: String?
    let email: String?
    let phone: String?
    let address: String?
    let discount_rate: Double?
    let is_loyal: Bool?
    let notes: String?
}

struct CustomerUpdateRequest: Encodable {
    let name: String
    let type: String?
    let email: String?
    let phone: String?
    let address: String?
    let discount_rate: Double?
    let is_loyal: Bool?
    let notes: String?
}

struct CustomerListItem: Identifiable {
    let id: Int
    let name: String
    let clientCode: String
    let phone: String
    let email: String
    let type: String
    let isLoyal: Bool

    var typeLabel: String {
        switch type {
        case "school": "École"
        case "student": "Étudiant"
        case "company": "Société"
        default: "Particulier"
        }
    }
}

File: android/core/data/src/main/java/com/app2/core/data/remote/dto/CustomerDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CustomerDTO(
    val id: Int,
    val name: String,
    val type: String? = "particulier",
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("client_code") val clientCode: String? = null,
    @SerialName("discount_rate") val discountRate: Double? = null,
    @SerialName("is_loyal") val isLoyal: Int? = null,
    @SerialName("is_active") val isActive: Int? = null,
    val ice: String? = null,
    val notes: String? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class CustomerCreateRequest(
    val name: String,
    val type: String? = null,
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("client_code") val clientCode: String? = null,
    @SerialName("discount_rate") val discountRate: Double? = null
)

@Serializable
data class CustomerUpdateRequest(
    val name: String? = null,
    val type: String? = null,
    val email: String? = null,
    val phone: String? = null,
    val address: String? = null,
    @SerialName("discount_rate") val discountRate: Double? = null,
    @SerialName("is_active") val isActive: Int? = null
)
